packages/datetime/src/datePickerCore.tsx

```diff
@@ -17,14 +17,40 @@ export interface IDatePickerModifiers {
     [name: string]: (date: Date) => boolean;
 }
 
+export interface IDateFormatProps {
+    /**
+     * Date format string, passed to `formatDate` and `parseDate`.
+     */
+    format?: string;
+
+    /**
+     * Placeholder text to display in empty input fields.
+     * Recommended practice is to indicate the expected date format.
+     */
+    placeholder?: string;
+
+    /**
+     * Function to serialize a JavaScript `Date` to a string.
+     * The special value `null` indicates the absence of a date.
+     */
+    formatDate(date: Date | null, format?: string, locale?: string): string;
+
+    /**
+     * Function to deserialize a string to a JavaScript `Date` object.
+     * Return `false` if the string is an invalid date.
+     * Return `null` to represent the absence of a date.
+     */
+    parseDate(str: string, format?: string, locale?: string): Date | false | null;
```

Comment:
not a huge blocker, but curious about this API decision: why would a consumer expect `format` in this function? if they're providing `format` as a prop, they already know the format they're interested in. in the usage of `props.parseDate()` it seems like you're passing `props.format` directly anyway.

Reply:
prior art: http://react-day-picker.js.org/api/DayPickerInput#formatDate.

this is the only place the `format` prop gets used, so we could drop it from the API and push it to the user. but the nice thing about this usage is that only one set of moment utils for `format/parseDate` could be used for all formats.